### PR TITLE
Hellow World app with server communication

### PR DESCRIPTION
**Pull request by tahir24434**
Signed-off-by: tahir24434 tahir.rauf1@gmail.com

### CODE REVIEW

File: src/hw_test.py [hunk 1]
```diff
@@ -27,6 +33,14 @@ def run_test(self):
         # Launch HelloWorld client
         self.launch_hw_client()
 
+        # Cereate a HAnalyzer class with that IP/PORT/Taskid
+        hwsa = HWSrvAnalyser(self.hw_server_task_ip, self.hw_server_task_port, self.hw_server_app_id)
+        # Call a function "starttest" on the HAnalyzer class instance
+        hwsa.start_test()
+
+        # Call teststartfoo on HAnalyzer class instance
+        hwsa.do_req_resp('helloworld', tout_60s, arg1='AWSOME')
```
Comment (sushilks) [inline src/hw_test.py]: Checks are still missing on count.
Rest looks ok.

File: src/hw_test.py [hunk 2]
```diff
@@ -1,15 +1,21 @@
 #!/usr/bin/env python
-
-import time
-
+from sys import path
+path.append("hydra/src/main/python")
 from hydra.lib.runtestbase import RunTestBase
```
Comment (sushilks) [inline src/hw_test.py]: change to HydraBase

File: src/hw_test.py [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
+class HWSrvAnalyser(HAnalyser):
+    def __init__(self, server_ip, server_port, task_id):
+        HAnalyser.__init__(self, server_ip, server_port, task_id)
 
 
 class HW(RunTestBase):
     def __init__(self):
         self.config = ConfigParser()  # Object contains 'configurations' parsed from hydra.ini config file
-        RunTestBase.__init__(self, test_name='HelloWorld', options=None, app_dirs=['src'])
+        RunTestBase.__init__(self, test_name='HelloWorld', options=None, app_dirs=['src', 'target', 'hydra'])
```
Comment (sushilks) [inline src/hw_test.py]: Change to HydraBase

File: src/hw_test.py [hunk 4]
```diff
@@ -1,68 +1,102 @@
...
-        RunTestBase.__init__(self, test_name='HelloWorld', options=None, app_dirs=['src'])
-        self.hw_server_app_id = self.format_appname("/hw-server")
-        self.hw_client_app_id = self.format_appname("/hw-client")
-        self.hw_server_task_ip = None
-        self.hw_server_task_port = None
-        self.hw_client_task_ip = None
-        self.hw_client_task_port = None
-        self.add_appid(self.hw_server_app_id)
-        self.add_appid(self.hw_client_app_id)
+        RunTestBase.__init__(self, test_name='HelloWorld', options=None, app_dirs=['src', 'hydra'])
```
Comment (sushilks) [inline src/hw_test.py]: change to HydraBase

File: src/hw_test.py [hunk 5]
```diff
@@ -1,68 +1,102 @@
...
 
 
 class RunTest(object):
     def __init__(self):
         r = HW()
         r.start_appserver()
         r.run_test()
-        time.sleep(60)
+        print ("Communicating sendmsg signal to pub")
+        r.hwpa.do_req_resp('sendmsg', tout_60s, arg1='3')
```
Comment (sushilks) [inline src/hw_test.py]: why not arg1=3 
why does it need a string ?

File: src/hw_test.py [hunk 6]
```diff
@@ -1,68 +1,102 @@
...
+        print ("Communicating getstats signal to pub")
+        time.sleep(50)
```
Comment (sushilks) [inline src/hw_test.py]: sleep is not good please remove.

File: src/hw_test.py [hunk 7]
```diff
@@ -1,68 +1,102 @@
...
+        r.hwsa.do_req_resp('getstats', tout_60s)
```
Comment (sushilks) [inline src/hw_test.py]: instead do a repeated poll every second with a timeout of 60 second.

File: src/hw_pub.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    context = zmq.Context()
+    socket = context.socket(zmq.PUB)
+    print("ZMQ HWM is set to : %s" % socket.get_hwm())
+    socket.bind("tcp://*:%s" % pub_port)
+
+    # Make ready the server to listen commands
+    pub_rep_port = os.environ.get('PORT0')
+    print ("Starting HelloWorld pub at port [%s]", pub_rep_port)
+    hd = HDHelloWorldPub(pub_rep_port)
+    hd.soc = socket
```
Comment (sushilks) [inline src/hw_pub.py:56]: It's generally not a good idea to manipulate a member variable of a class.

File: src/hw_pub.py [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+
+class HDHelloWorldPub(HDaemonRepSrv):
+    def __init__(self, port):
+        HDaemonRepSrv.__init__(self, port)
+        soc = None
+        self.register_fn('sendmsg', self.send_msg)
+
+    def send_msg(self, arg1):
+        l.info("send_msg has been called with argument %s" % arg1)
```
Comment (sushilks) [inline src/hw_pub.py:30]: arg1 is unused.

File: src/hw_test.py (hunk 7 above)
Comment (sushilks) [inline src/hw_test.py]: Also the value returned by 'getstats' is not being checked.

File: src/hw_test.py (hunk 1 above)
Comment (tahir24434) [inline src/hw_test.py]: Done.

File: src/hw_test.py (hunk 2 above)
Comment (tahir24434) [inline src/hw_test.py]: Done.

File: src/hw_test.py (hunk 3 above)
Comment (tahir24434) [inline src/hw_test.py]: Done.

File: src/hw_test.py (hunk 4 above)
Comment (tahir24434) [inline src/hw_test.py]: Done.

File: src/hw_test.py (hunk 5 above)
Comment (tahir24434) [inline src/hw_test.py]: Done.

File: src/hw_test.py (hunk 6 above)
Comment (tahir24434) [inline src/hw_test.py]: Removed.

File: src/hw_test.py (hunk 7 above)
Comment (tahir24434) [inline src/hw_test.py]: Done.

File: src/hw_pub.py (hunk 9 above)
Comment (tahir24434) [inline src/hw_pub.py:30]: Removed.

**Pull request merged**